src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/config/ControllerAdvice.kt

```diff
@@ -89,6 +90,20 @@ class ControllerAdvice {
       )
   }
 
+  @ExceptionHandler(HttpMessageNotReadableException::class)
+  fun handleRequestUnreadableException(e: HttpMessageNotReadableException): ResponseEntity<ErrorResponse> {
+    log.info("Message not readable exception: {}", e.message)
+    return ResponseEntity
+      .status(HttpStatus.BAD_REQUEST)
+      .body(
+        ErrorResponse(
+          status = HttpStatus.BAD_REQUEST.value(),
+          userMessage = "Bad request: ${e.message}",
+          developerMessage = e.message
+        )
+      )
+  }
+
   @ExceptionHandler(ValidationException::class, MethodArgumentNotValidException::class)
```

Comment:
You could reuse the same controller advice by adding the exception type to this list, not sure if you want to do that

Reply:
Consolidate on next PR?